src/Workspaces/CSharp/Portable/Utilities/CSharpUseImplicitTypeHelper.cs

```diff
@@ -259,6 +259,11 @@ protected override bool ShouldAnalyzeForEachStatement(ForEachStatementSyntax for
                 return false;
             }
 
+            if (!IsSwitchExpressionAndCanUseVar(typeName, initializer, semanticModel))
+            {
+                return false;
+            }
```

Comment:
this reads very strangely in english.  "if this is not a switch expressoin and [we] can use var ... then don't allow var".  What is likely better is:

```
if (IsSwitchExpressionAndCannotUseVar(...))
    return false;
```